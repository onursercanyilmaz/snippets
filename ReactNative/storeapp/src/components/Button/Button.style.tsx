import {Dimensions, StyleSheet} from 'react-native';

export default {
  primary: StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      backgroundColor: '#be2945',
      borderRadius: 50,
      alignItems: 'center',
      width: Dimensions.get('window').width - 200,
      alignContent: 'center',
      alignSelf: 'center',
      marginTop: 20,
      marginBottom: 20,
    },

    label: {
      color: '#fff',
      fontSize: 20,
      fontWeight: 'bold',
      marginBottom: 10,
      textAlign: 'center',
    },
  }),

  secondary: StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      backgroundColor: '#fff',
      borderRadius: 50,
      alignItems: 'center',
      width: Dimensions.get('window').width - 200,
      alignContent: 'center',
      alignSelf: 'center',
      marginTop: 20,
      marginBottom: 20,
    },

    label: {
      color: '#be2945',
      fontSize: 20,
      fontWeight: 'bold',
      marginBottom: 10,
      textAlign: 'center',
    },
  }),
};
