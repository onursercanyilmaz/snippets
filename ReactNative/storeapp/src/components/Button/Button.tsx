import {
  View,
  Text,
  Touchable,
  TouchableOpacity,
  ActivityIndicator,
} from 'react-native';
import React from 'react';
import styles from './Button.style';

interface ButtonProps {
  onPress: (() => void) | undefined;
}
export default function Button(props: ButtonProps) {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={props.onPress}
      testID='button'
    >
      <Text style={styles.label}>Login</Text>
    </TouchableOpacity>
  );
}
