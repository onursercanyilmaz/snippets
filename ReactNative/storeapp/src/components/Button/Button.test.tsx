import React from 'react';
import {render} from '@testing-library/react-native';
import Button from './Button';
//npm test
test('', () => {
  const component = render(<Button onPress={undefined} />);
  expect(component).toMatchSnapshot();
});
