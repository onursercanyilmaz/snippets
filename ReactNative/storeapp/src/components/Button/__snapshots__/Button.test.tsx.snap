// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#be2945",
      "borderRadius": 50,
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 20,
      "marginTop": 20,
      "opacity": 1,
      "width": 550,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Login
  </Text>
</View>
`;
